src/VerletList.cpp

```diff
@@ -84,25 +86,180 @@ namespace espressopp {
   
   void VerletList::rebuild()
   {
+    timer.reset();
+    real currTime = timer.getElapsedTime();
+
     //real cutVerlet = cut + getSystem() -> getSkin();
     cutVerlet = cut + getSystem() -> getSkin();
     cutsq = cutVerlet * cutVerlet;
     
     vlPairs.clear();
 
-    // add particles to adress zone
-    CellList cl = getSystem()->storage->getRealCells();
-    LOG4ESPP_DEBUG(theLogger, "local cell list size = " << cl.size());
-    for (CellListAllPairsIterator it(cl); it.isValid(); ++it) {
-      checkPair(*it->first, *it->second);
-      LOG4ESPP_DEBUG(theLogger, "checking particles " << it->first->id() << " and " << it->second->id());
+    if(useBuffers) {
```

Comment:
Could you check if this `if (useBuffers)` has some impact on performance in case of `useBuffers = true|false` and if you modify the code by removing if and keeping only `rebuildUsingBuffers` statement.

Reply:
This does not impact performance since the branch is only reached once every rebuild. It doesn't take a lot of time compared to the more critical loops under it. I also checked with VTune and the branch is not marked as a hotspot.